Parser/pegen.c

```diff
@@ -471,7 +473,7 @@ _PyPegen_raise_error_known_location(Parser *p, PyObject *errtype,
            we're actually parsing from a file, which has an E_EOF SyntaxError and in that case
            `PyErr_ProgramTextObject` fails because lineno points to last_file_line + 1, which
            does not physically exist */
-        assert(p->tok->fp == NULL || p->tok->fp == stdin || p->tok->done == E_EOF);
+        assert(p->tok->fp == NULL || p->tok->fp == stdin || p->tok->done == E_EOF || !uses_utf8_codec);
```

Comment:
Could you amend the comment above this line to explain this additional encoding edge case?

Reply:
Sure

Reply:
Aah in that case a comment is probably better above the `int uses_utf8_codec = (!p->tok->encoding || strcmp(p->tok->encoding, "utf-8") == 0);` line explaining that we need it because `PyErr_ProgramTextObject` assumes the file is utf-8.

Reply:
Makes sense